src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AccountService.java

```diff
@@ -301,4 +307,24 @@ private boolean handleAccountCreationEmail(AzureAccount createdAccount, String f
         return isSuccessful;
     }
 
+    /**
+     * Delete a user account by the supplied email.
+     * This deletes the user from AAD, our user table and subscriptions.
+     *
+     * @param email The email of the user to delete.
+     * @return Confirmation message that account has been deleted.
+     */
+    public String deleteAccount(String email) {
+        PiUser userToDelete = userRepository.findByEmail(email)
+            .orElseThrow(() -> new NotFoundException("User with supplied email could not be found"));
+        try {
+            azureUserService.deleteUser(userToDelete.getProvenanceUserId());
+            log.info(subscriptionService.sendSubscriptionDeletionRequest(userToDelete.getUserId().toString()));
+            userRepository.delete(userToDelete);
+        } catch (AzureCustomException ex) {
+            log.error("Error when deleting an account from azure with Provenance user id: %s and error: %s",
+                      userToDelete.getProvenanceUserId(), ex.getMessage());
```

Comment:
This will return "User with ID %s has been deleted" when the user has not been deleted fully. I reckon you need to amend this.

Reply:
Will only return that message if it deletes now :)